app/src/shared/web3/Web3Poller.js

```diff
@@ -49,6 +49,10 @@ class CancelError extends Error {
     }
 }
 
+const allowedCaller = {}
```

Comment:
@tumppi, here's an interesting pattern. I use `caller` vs `allowedCaller` in the constructor, this new instances of `Web3Poller` can only be created from within the file.

Maybe something you'll find interesting.